faunadb/client.py

```diff
@@ -42,36 +68,49 @@ def __init__(
       Port of the FaunaDB server.
     :param timeout:
       Read timeout in seconds.
-    :param secret:
-      Auth token for the FaunaDB server.
     :param observer:
       Callback that will be passed a :any:`RequestResult` after every completed request.
+    :param pool_connections:
+      The number of connection pools to cache.
+    :param pool_maxsize:
+      The maximum number of connections to save in the pool.
     """
 
     self.domain = domain
     self.scheme = scheme
     self.port = (443 if scheme == "https" else 80) if port is None else port
 
-    self.session = Session()
-    if secret is not None:
-      self.session.auth = (secret, "")
-
-    self.session.headers.update({
-      "Accept-Encoding": "gzip",
-      "Content-Type": "application/json;charset=utf-8"
-    })
-    self.session.timeout = timeout
-
+    self.auth = HTTPBasicAuth(secret, "")
     self.base_url = "%s://%s:%s" % (self.scheme, self.domain, self.port)
-
     self.observer = observer
 
+    self.pool_connections = pool_connections
+    self.pool_maxsize = pool_maxsize
+
+    if ('session' not in kwargs) or ('counter' not in kwargs):
+      self.session = Session()
+      self.session.mount('https://', HTTPAdapter(pool_connections=pool_connections,
+                                                 pool_maxsize=pool_maxsize))
+      self.session.mount('http://', HTTPAdapter(pool_connections=pool_connections,
+                                                pool_maxsize=pool_maxsize))
+      self.counter = _Counter(1)
+
+      self.session.headers.update({
+        "Accept-Encoding": "gzip",
+        "Content-Type": "application/json;charset=utf-8"
+      })
+      self.session.timeout = timeout
+    else:
+      self.session = kwargs['session']
+      self.counter = kwargs['counter']
+
   def __del__(self):
     # pylint: disable=bare-except
-    try:
-      self.session.close()
-    except:
-      pass
+    if self.counter.decrement() == 0:
+      try:
```

Comment:
yeah, I really wanted to remove this but I was afraid to do, though, I removed and no exception was thrown